front/src/modules/utils/cookie-storage.ts

```diff
@@ -0,0 +1,62 @@
+import Cookies, { CookieAttributes } from 'js-cookie';
+
+type Listener = (
+  newValue: string | undefined,
+  oldValue: string | undefined,
+) => void;
+
+class CookieStorage {
```

Comment:
Could we try to implement this with hooks ? I don't know how introducing classes will scale in a functional React / Recoil ecosystem.